oeps/oep-0010.rst

```diff
@@ -0,0 +1,202 @@
+will be where changes are accumulated to create each release in the line.
+Releases will be tagged "open-release/RELEASENAME.1",
+"open-release/RELEASENAME.2", and so on.
+
+
+Involving repos in the Open edX build process
+---------------------------------------------
+
+:doc:`OEP-2 <oep-0002>` defines a file format for repository metadata.  The 
+``openedx-release`` key is an optional dictionary governing the participation
+of the repo in the Open edX release process.
+
+``openedx-release``: dictionary (optional)
+
+    Possible keys:
+
+    ``ref``
+        The name of the release-from branch in this repo. This is the branch
+        that will be tagged when an Open edX release is made.
+
+    ``requirements``
+        The path to a requirements.txt-style file in this repo. This file will
+        be consulted to find the commit to tag in dependent repos indicated by
+        ``parent-repo`` keys.
+
+    ``parent-repo``
+        The name of another repo, one that installs this repo via a
+        requirements file.  The parent repo's requirements file will indicate
+        the commit to tag in this repo.
+
+        .. note::
+            We believe we no longer need to tag dependent repos, so this may go
+            away.
+
+
+
+Installing repos
+----------------
+
+Our installations are based on Anisble playbooks.  Up until the Eucalyptus
+release, all three supported installation methods were single-machine: all of
+the Open edX software was installed and ran on a single machine, either a
+Virtualbox image, or a native machine.
+
+That model does not scale up as the number of services and applications grows.
+Newer services are supporting Docker for installation.  Eventually, we would
+like to be in an all-Docker world where an installation is just a constellation
+of Docker containers.
+
+To allow us to move gradually from a single-machine model to an all-Docker
+model, we'll support a machine running a number of edX services and
+applications, and also running a number of Docker containers.
+
+
+Installation methods
+--------------------
+
+There are three supported installation methods:
+
+- Devstack: for developers planning to modify Open edX code.
+
+- Native: for adopters looking to experiment with Open edX in a non-production
+  environment.  Also useful as a starting point for people designing their
+  production deployment.
+
+- Fullstack: same as Native, but in a pre-packaged Vagrant image.
+
+The devstack installation will follow :doc:`OEP-5 <oep-0005>`, with a
+single-machine Vagrant image running alongside some number of Docker
+containers.
+
+.. note::
+
+   We haven't determined how best to allow developers to configure which
+   services to run and which should be editable.
+
+The native and fullstack installation will use Ansible playbooks to install
+Open edX components onto the machine, and install Docker and some number of
+Docker containers.
```

Comment:
It seems like it would be good to be specific about the Ansible playbook. Right now, as an Open edX developer, I don't have enough information to determine for myself whether a given thing is installed in an Open edX release or not.